pkg/repository/ghreleases/ghreleases.go

```diff
@@ -82,9 +82,54 @@ func (ghr *GHReleases) PublishProbe(driverVersion string, probePath string) erro
 }
 
 // IsAlreadyMirrored implmements repository.Repository.IsAlreadyMirrored for GitHub Releases.
-func (ghr *GHReleases) IsAlreadyMirrored(driverVersion string, probeName string) bool {
-	// TODO: unimplimented
-	return false
+func (ghr *GHReleases) IsAlreadyMirrored(driverVersion string, probeName string) (bool, error) {
+	// use a mutex to ensure that this function is only called once at a time as it is not thread safe.
+	// i.e. a release that doesn't exist may result in multiple goroutines trying to create it at once.
+	ghr.releasesMu.Lock()
+	defer ghr.releasesMu.Unlock()
+	ctx := context.Background()
+
+	// Retrieve the releases
```

Comment:
There's a lot of nesting going on under here. Try thinking about how you could break this into smaller functions to reduce that nesting. e.g.
```go
release, err := ghr.getReleaseByName(driverVersion)
asset, err := ghr.getAssetFromReleaseByName(release, probeName)
```

Reply:
I've split these 2 out into their own functions as suggested.